main.php

```diff
@@ -33,3 +33,24 @@
 
 $entityManager->persist($category);
 $entityManager->flush();
+
+
+
+// TEST 2
+$data = [
+    'title' => 'Test Product',
+    'description' => '...',
+    'category' => [
+        'id' => $category->getId()
+    ]
+];
+$jsonData = json_encode($data);
+$product = $serializer->deserialize($jsonData, 'Product', 'json');
+
+// var_dump($product);
+// in this case category object should be loaded, that is the problem
+// it works if we set
+// $product->setCategory($category);
+
+$entityManager->persist($product);
+$entityManager->flush();
```

Comment:
PHP Fatal error:  Uncaught exception 'Doctrine\ORM\ORMInvalidArgumentException' with message 'A new entity was found through the relationship 'Product#category' that was not configured to cascade persist operations for entity: Category@000000001167c3e3000000016fb5647b. To solve this issue: Either explicitly call EntityManager#persist() on this unknown entity or configure cascade persist  this association in the mapping for example @ManyToOne(..,cascade={"persist"}). If you cannot find out which entity causes the problem implement 'Category#__toString()' to get a clue.' in /Users/ivan/www/jms-serializer-doctrine-standalone-sample/vendor/doctrine/orm/lib/Doctrine/ORM/ORMInvalidArgumentException.php:91
Stack trace: